internal/controller/olsconfig_controller.go

```diff
@@ -32,34 +35,48 @@ import (
 type OLSConfigReconciler struct {
 	client.Client
 	Scheme *runtime.Scheme
+	logger logr.Logger
 }
 
 //+kubebuilder:rbac:groups=ols.openshift.io,resources=olsconfigs,verbs=get;list;watch;create;update;patch;delete
 //+kubebuilder:rbac:groups=ols.openshift.io,resources=olsconfigs/status,verbs=get;update;patch
 //+kubebuilder:rbac:groups=ols.openshift.io,resources=olsconfigs/finalizers,verbs=update
+// RBAC for managing deployments of OLS application server
+//+kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
```

Comment:
Is this creating clusterrole ? I feel just the role should be enough we should operate in a namespace

Reply:
operator watches resources that can be created in any namespace so we need ClusterRole